Tests/FluentAssertions.Specs/Common/MemberInfoExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Reflection;
+
+namespace FluentAssertions.Specs.Common
+{
+    internal static class MemberInfoExtensions
+    {
+        public static Type GetUnderlyingType(this MemberInfo member)
+        {
+            switch (member.MemberType)
```

Comment:
A good candidate for a switch expression

```cs
return member.MemberType switch
{
    MemberTypes.Event => ((EventInfo)member).EventHandlerType,
    MemberTypes.Field => ((FieldInfo)member).FieldType,
    MemberTypes.Method => ((MethodInfo)member).ReturnType,
    MemberTypes.Property => ((PropertyInfo)member).PropertyType,
    _ => throw new ArgumentException("Input MemberInfo must be if type EventInfo, FieldInfo, MethodInfo, or PropertyInfo");
};
```

Reply:
![image](https://user-images.githubusercontent.com/572734/97480471-bedbfc00-1953-11eb-95fe-c6f7d8bd75bc.png)

Awesome...